components/layout/DocsLayout.js

```diff
@@ -16,13 +16,23 @@ import AnnouncementHero from '../campaigns/AnnoucementHero'
 import { SearchButton, DOCS_INDEX_NAME } from '../AlgoliaSearch';
 import IconLoupe from '../icons/Loupe';
 import { getAllPosts } from '../../lib/api'
+import editOptions from '../../config/edit-page-config.json'
 
 function generateEditLink(post) {
-  if (post.slug.includes('/specifications/')) {
-    return <a target="_blank" rel="noopener noreferrer" href={`https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md`} className="ml-1 underline">Edit this page on GitHub</a>
-  } 
-  return <a target="_blank" rel="noopener noreferrer" href={`https://github.com/asyncapi/website/blob/master/pages${post.isIndex ? post.slug + '/index' : post.slug}.md`} className="ml-1 underline">Edit this page on GitHub</a>
-}
+
+  var last=post.id.substring(post.id.lastIndexOf("/") + 1);
+  var target=editOptions.find(edit=>{ return post.slug.includes(edit.value)});
+  
+ 
+
+  if(target.value==""){
+    return <a target="_blank" rel="noopener noreferrer" href={`${target.href}${post.isIndex ? post.slug + '/index' : post.slug}.md`} className="ml-1 underline">Edit this page on GitHub</a>
+
+  }
+  return <a target="_blank" rel="noopener noreferrer" href={`${target.href}/${target.value=="reference/specification/" ?'asyncapi.md':last}`} className="ml-1 underline">Edit this page on GitHub</a>
```

Comment:
can we add logic that is not forcing us to do hardcode like `${target.value=="reference/specification/" ?'asyncapi.md':last}` ?

so support:
```
{
        "value": "reference/specification/",
        "href": "https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md"
    }
```

and you check, if the last element of the `href` is a filename, then `Edit the page on GitHub` always point to it on pages from `reference/specification/"`